### PR TITLE
RFC: Simplify quickstart and include http event 

### LINKED ISSUE #3708: Simplify quickstart and include http event
# Docs / Feature Proposal

## Description

The current [quickstarts](https://serverless.com/framework/docs/providers/aws/guide/quick-start/
) suggests to use the template `aws-nodejs`. Which has two drawbacks:

a) It doesn't show an end2end experience. 
b) Has a bloated serverless.yml. Even if it's just the comments.

## Proposal

I would suggest using the `hello-world` template in the quickstart including the http get event. As it

a) Shows the simplest end2end experience. Including an event.
b) Has the most simple serverless.yml without overwhelming the user. Additionally the cors code could be removed.

### PR DESCRIPTION
**Pull request by luebken**
## What did you implement:

Closes #3708

* Using `hello-world` as the template in aws quickstart
* Simplifying hello-world further

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** NO
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/create/templates/hello-world/serverless.yml [hunk 1]
```diff
@@ -21,4 +21,3 @@ functions:
       - http:
           path: hello-world
           method: get
-          cors: true
```
Comment (pmuens) [inline lib/plugins/create/templates/hello-world/serverless.yml]: This was added by @DavidWells so that CORS is always in place and it's easier to start with the `hello-world` template.

File: lib/plugins/create/templates/hello-world/handler.js [hunk 2]
```diff
@@ -3,9 +3,6 @@
 module.exports.helloWorld = (event, context, callback) => {
   const response = {
     statusCode: 200,
-    headers: {
```
Comment (pmuens) [inline lib/plugins/create/templates/hello-world/handler.js]: This should be kept so that CORS works.

File: lib/plugins/create/create.js [hunk 3]
```diff
@@ -21,6 +21,7 @@ const validTemplates = [
   'openwhisk-python',
   'openwhisk-swift',
   'google-nodejs',
+  'hello-world',
```
Comment (pmuens) [inline lib/plugins/create/create.js]: This is already added below (after the `plugin` definition) 🔽

File: lib/plugins/create/templates/hello-world/serverless.yml (hunk 1 above)
Comment (luebken) [inline lib/plugins/create/templates/hello-world/serverless.yml]: Could you or @DavidWells elaborate why? All my tests worked just fine without CORS enabled.

I do think it is great to have a nice simple way to enable CORS and show that. But if we show it we should also explain why and have the example / test highlight that feature. IMHO a helloworld should be as small as possible. So I would leave CORS out.

File: lib/plugins/create/create.js (hunk 3 above)
Comment (luebken) [inline lib/plugins/create/create.js]: uupps. sorry about that.

File: lib/plugins/create/templates/hello-world/serverless.yml (hunk 1 above)
Comment (pmuens) [inline lib/plugins/create/templates/hello-world/serverless.yml]: @luebken Sure. The main motivation was that the `http` event definition w/o CORS enabled is too basic for "serious" applications.

We also elaborated if we should enable CORS by default whenever you use the `http` event but thought that this might be too much for the first iteration.

More information about the decisions / motivation for the `hello-world` template can be found here: https://github.com/serverless/serverless/pull/3445

Comment (luebken) [inline lib/plugins/create/templates/hello-world/serverless.yml]: Thanks for the comments. But it's a little hard to follow the other issues / PRs.

Let me see if I get your motivation for CORS correctly: "If you expose a function via an http endpoint this will likely be consumed by some frontend javascript. Which will likely be served from another endpoint and therefor result in CORS problems."

If that is the case than we should show that in the hello-world example and serve some frontend javascript and with this motivate CORS. 

I would be in the camp of keeping the http enabled hello-world as simple as possible and in a second step (e.g. e.g. hello-html-world) add CORS and motivate it.

(Sorry for making this debate so big about a small change. I'm just stating my thoughts. Feel free to close / pause this PR.)

**Comment by pmuens** [comment]
Thanks again for opening and discussing this @luebken 👍 

I'll close this PR for now but we might get back to it since we're currently re-working the onboarding experience and therefore our templates and docs.

Thanks for taking the time to improve Serverless 👍

**Comment by luebken** [comment]
Sure. No worries.

**Comment by luebken** [comment]
And BTW: the more I play around with serverless the more I think helloworld shouldn't contain cors. ;-) Just my 2 € cent.

**Comment by pmuens** [comment]
> Sure. No worries.

Thanks!

> And BTW: the more I play around with serverless the more I think helloworld shouldn't contain cors. ;-) Just my 2 € cent.

Thanks for the feedback. We'll take that into consideration when refactoring the templates and docs 👍
